platform-services.md

```diff
@@ -0,0 +1,215 @@
+sample project as the starter code to build your own Flutter app.
+
+The following diagram shows how the `hello_services` app uses the `FlutterView` interface to
+send and receive messages to platform and third-party services on iOS and Android.
+
+ ![hello_services architecture](/images/hello_services.png)
+
+**Fig 1.** Architecture of the `hello_services` sample.
+
+The Dart, iOS, and Android-specific components are related as follows:
+
+* In the Dart code, the sample app imports the
+[`flutter/services.dart`](http://docs.flutter.io/flutter/services/services-library.html)
+package. This package defines the
+[`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html) class,
+which provides a bridge to send and listen for JSON-encoded messages exchanged with a platform
+service on a <em>host</em> app (that is, the iOS or Android-specific parts of your app). The
+<code>hello_services</code> app uses the static methods of the
+[`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html) class
+in the [`State`](http://docs.flutter.io/flutter/material/State-class.html) object of a
+stateful widget to make services requests, then updates the UI accordingly.
+
+* On iOS, the sample app imports the Flutter framework library (`Flutter.h`) as a
+dependency. The library defines the <code>FlutterViewController</code> class, which is a
+[`UIViewController`](https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIViewController_Class/)
+that provides a bridge from the
+[`UIKit`](https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIKit_Framework/)
+view hierarchy to the Flutter app. The `hello_services` app sets an instance of
+[`FlutterViewController`](https://github.com/flutter/engine/blob/master/sky/shell/platform/ios/framework/Headers/FlutterViewController.h)
+as the root view controller of its [`UIWindow`](https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIWindow_Class/).
+
+* On Android, the sample imports the `flutter.jar` file as a dependency. This JAR
+defines the [`FlutterView`](https://github.com/flutter/engine/blob/master/sky/shell/platform/android/io/flutter/view/FlutterView.java)
+class, which is an [`android.view.View`](https://developer.android.com/reference/android/view/View.html)
+subclass that provides a bridge from the Android view hierarchy to the Flutter app. The
+`hello_services` app creates an instance of [`FlutterView`](https://github.com/flutter/engine/blob/master/sky/shell/platform/android/io/flutter/view/FlutterView.java)
+in its main activity.
+
+### App structure {#app-structure}
+
+The `hello_services` sample app has the following project file structure.
+
+ ![hello_services sample files](/images/hello_services_files.png)
+
+**Fig. 2.** Project file structure for the `hello_services` sample.
+
+To make platform-specific service calls, you may need to modify these key files and folders in the `hello_services` project:
```

Comment:
Do you mean "to create additional platform-specific services" ?  To make a call, you need only invoke one of the existing interfaces.

Also, I'd strike the word "key".  It doesn't meany anything in this content.